Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+      return false;
+    };
+
+    for (let i = 0; i < itemInfos.length; i++)
+    {
+      var item = DropDownMenu.CreateItem(itemInfos[i], selectionCount);
+      if (item != null)
+        ul.appendChild(item);
+    }
+
+    var titleDivFunc = DropDownMenu.CreateTitleDivGetter (context);
+    DropDownMenu.ApplyPosition ($(div), evt, titleDivFunc());
+    $(div).iFrameShim({ top: '0px', left: '0px', width: '100%', height: '100%' });
+
+    if (DropDownMenu._repositionTimer) 
+      clearTimeout(DropDownMenu._repositionTimer);
+    var repositionHandler = function ()
+    {
+      if (DropDownMenu._repositionTimer)
+        clearTimeout (DropDownMenu._repositionTimer);
+
+      if (DropDownMenu._currentPopup && DropDownMenu._currentPopup == div && $(div).is (':visible'))
+      {
+        DropDownMenu.ApplyPosition ($(div), null, titleDivFunc());
+        DropDownMenu._repositionTimer = setTimeout (repositionHandler, DropDownMenu._repositionInterval);
+      }
+    };
+
+    // Only reposition if opened via titleDiv
+    if (evt == null)
+      DropDownMenu._repositionTimer = setTimeout(repositionHandler, DropDownMenu._repositionInterval);
+
+    if ($menuButton.data(DropDownMenu._button_timestampDataKey))
+    {
+      $menuButton.removeData(DropDownMenu._button_timestampDataKey);
+    }
+    else
+    {
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+    }
+  }
+
+  private static CreateTitleDivGetter ($context: JQuery): DropDownMenu_TitleDivGetter
+  {
+    const context = $context[0];
+    const contextID = context.id;
+    if (contextID == null)
+    {
+      return function ()
+      {
+        // TODO: nullcheck?
+        return $(context.firstElementChild!);
+      };
+    }
+    else
+    {
+      return function ()
+      {
+        // TODO: shouldn't this be rewritten anyway? getElementById should only be called once here.
+        // TODO: nullability
+        return document.getElementById(contextID)! && $(document.getElementById(contextID)!.firstElementChild!);
+      };
+    }
+  }
+
+  private static ApplyPosition (popUpDiv: JQuery, clickEvent: Nullable<JQueryMouseEventObject>, referenceElement: JQuery): void
+  {
+    var space_top = Math.round(referenceElement.offset().top - $(document).scrollTop());
+    var space_bottom = Math.round($(window).height() - referenceElement.offset().top - referenceElement.height() + $(document).scrollTop());
+    var space_left = referenceElement.offset().left;
+    var space_right = $(window).width() - referenceElement.offset().left - referenceElement.width();
+
+    // position drop-down list
+    var top = clickEvent ? clickEvent.clientY : Math.max(0, space_top + referenceElement.outerHeight());
+    var left = clickEvent ? clickEvent.clientX : 'auto';
+    var right = clickEvent ? 'auto' : Math.max(0, $(window).width() - referenceElement.offset().left - referenceElement.outerWidth());
+
+    popUpDiv.css('top', top);
+    popUpDiv.css('bottom', 'auto');
+    // TODO: Missing unit
```

Comment:
bug ticket. there might already be one for missing units

Reply:
https://re-motion.atlassian.net/browse/RM-7699 should this be added here?

Reply:
yes, and the ticket reworded. I think it's okay to fix all the PX errors in one ticket.

Reply:
@a-ctor The fix you provided with `'' + right` does not work here.
The following JQuery types are defined:
![image](https://user-images.githubusercontent.com/40357550/111319433-478ad880-8666-11eb-9301-a27274e6eadb.png)
If a string without `px` is passed in, the outcoming setting is wrong - in my case, the dropdown menu popup is stuck to the right side of the browser. If a number is passed in, the code is working fine. I chose to do `popUpDiv.css('right', right as number);` now, as this satisfies the transpiler and the code is still working fine.

cc @MichaelKetting @Stefan-Ilic